### ISSUE TITLE
super inside eval inside functions should be an early error

### ISSUE THREAD

**Issue by bterlson** [Microsoft/ChakraCore#1558]
``` js
function foo () {
    // these examples should be early errors (ie. we don't print 1).
    eval('print(1); super();');
    eval('print(1); super.foo();');
}

foo();

/*
┌─────────────────────┬─────────────────────────────────────────────────────────────────────┐
│ d8                  │                                                                     │
│                     │ SyntaxError: 'super' keyword unexpected here                        │
├─────────────────────┼─────────────────────────────────────────────────────────────────────┤
│ chakra-es6          │ 1                                                                   │
│ chakra-experimental │ ReferenceError: 'super' can only be accessed from a subclass method │
│ chakra              │                                                                     │
├─────────────────────┼─────────────────────────────────────────────────────────────────────┤
│ sm                  │                                                                     │
│                     │ SyntaxError: super() is only valid in derived class constructors:   │
├─────────────────────┼─────────────────────────────────────────────────────────────────────┤
│ jsc                 │                                                                     │
│                     │ SyntaxError: super is not valid in this context.                    │
└─────────────────────┴─────────────────────────────────────────────────────────────────────┘
*/
```

Our implementation is in line with the current spec but not (I believe) with the spec's intent or other implementations. I have a [262 PR](https://github.com/tc39/ecma262/pull/685) codifying this.

**Comment by dilijev** [comment]
[Consensus was reached](https://github.com/tc39/ecma262/pull/685#issuecomment-249950269)

**Comment by curtisman** [comment]
@suwc, is this fixed in 1f4e125564723fb87b7e5dfff043ee7af36393c5?

**Comment by suwc** [comment]
@curtisman, yes this has been fixed in `1f4e125`, a.k.a. https://github.com/Microsoft/ChakraCore/pull/2098
